File: src/main/scala/org/xbery/artbeams/categories/repository/CategoryFilter.scala
package org.xbery.artbeams.categories.repository

import java.time.Instant

import org.xbery.artbeams.common.assets.repository.{AssetFilter, ValidityAssetFilter}

/**
  * @author Radek Beran
  */
case class CategoryFilter(
  override val id: Option[String],
  override val ids: Option[Seq[String]],
  override val createdBy: Option[String],
  override val validityDate: Option[Instant],
  slug: Option[String],
  title: Option[String]) extends AssetFilter with ValidityAssetFilter

object CategoryFilter {
  lazy val Empty = CategoryFilter(None, None, None, None, None, None)
}

File: src/main/kotlin/org/xbery/artbeams/categories/repository/CategoryFilter.kt
package org.xbery.artbeams.categories.repository

import org.xbery.artbeams.common.assets.repository.AssetFilter
import org.xbery.artbeams.common.assets.repository.ValidityAssetFilter
import java.time.Instant

/**
 * @author Radek Beran
 */
data class CategoryFilter(
    override val id: String?,
    override val ids: List<String>?,
    override val createdBy: String?,
    override val validityDate: Instant?,
    val slug: String?,
    val title: String?
) : AssetFilter, ValidityAssetFilter {
    companion object {
        val Empty: CategoryFilter = CategoryFilter(null, null, null, null, null, null)
    }
}